### PR TITLE
141 serializable metrics

### PR DESCRIPTION
**Pull request by dominikmaeckel**

### CODE REVIEW

**Comment by tillschallau** [comment]
Let's discuss moving all the extension functions and their files into the `tools.aqua.stars.core.metric.serialization` package, or some subpackage. Maybe some of these files can also be merged.

**Comment by tillschallau** [comment]
We should also rename `DataSaver.kt` as this might be confusing. `PlottingDataSaver.kt` might be a valid option.

**Comment by tillschallau** [comment]
Still to do:

- [ ] Add `Serializable` interface to more metrics
- [ ] Add some more `SerializationResult`s while implementing `Serializable` for metrics
- [ ] Some more tests
- [ ] A final comparison of all results (creation of final results file for pipeline)
- [ ] Create ground-truth data for experiments
- [ ] Update pipeline to check for changed evaluations

**Comment by tillschallau** [comment]
#164 should be merged first.

**Comment by tillschallau** [comment]
Currently missing interface implementations:
- InvalidTSCInstances
- MissingTSCInstances
- PostEvaluationMetrics

File: stars-core/src/main/kotlin/tools/aqua/stars/core/metric/metrics/postEvaluation/MissedPredicateCombinationsPerTSCMetric.kt [hunk 1]
```diff
@@ -47,15 +50,18 @@ import tools.aqua.stars.core.types.*
...
     U : TickUnit<U, D>,
     D : TickDifference<D>>(
     override val dependsOn: ValidTSCInstancesPerTSCMetric<E, T, S, U, D>,
-    override val logger: Logger = Loggable.getLogger("missing-predicate-combinations")
-) : PostEvaluationMetricProvider<E, T, S, U, D>, Loggable {
+    override val logger: Logger = Loggable.getLogger("missed-predicate-combinations")
+) : PostEvaluationMetricProvider<E, T, S, U, D>, Serializable, Loggable {
+
+  /** Holds the evaluation result after calling [postEvaluate]. */
+  private var evaluationResult: Map<TSC<E, T, S, U, D>, Set<PredicateCombination>>? = null
```
Comment (dominikmaeckel) [inline stars-core/src/main/kotlin/tools/aqua/stars/core/metric/metrics/postEvaluation/MissedPredicateCombinationsPerTSCMetric.kt:64]: Consider renaming to evaluationResultCache

Comment (tillschallau) [reply stars-core/src/main/kotlin/tools/aqua/stars/core/metric/metrics/postEvaluation/MissedPredicateCombinationsPerTSCMetric.kt]: Resolved in 018a5bb4836705e27b19d0e43df1ff2ee0917457

**Review by tillschallau — APPROVED** [review]

**Pull request merged**